test/test_save_as_netcdf.py

```diff
@@ -78,91 +79,96 @@ def nc_var_has_attr_vals(ds, var_name, att_dict):
     for key, val in att_dict.items():
         try:
             if val != getattr(ds.variables[var_name], key):
-                print("attribute:", key)
-                print("expected val:", val)
-                print("val in file:", repr(getattr(ds.variables[var_name], key)))  # noqa
                 return False
         except AttributeError:
-            print(key, "is not an attribute of var:", var_name)
             return False
     return True
 
 
 def test_simple_write():
-    fname = 'temp.nc'
     grid = two_triangles()
 
-    grid.save_as_netcdf(fname)
+    fname = 'temp.nc'
+    with chdir(test_files):
+        grid.save_as_netcdf(fname)
+        ds = netCDF4.Dataset(fname)
+        os.remove(fname)
 
-    # Could be lots of tests here.
-    with netCDF4.Dataset(fname) as ds:
-        assert nc_has_variable(ds, 'mesh')
-        assert nc_var_has_attr_vals(ds, 'mesh', {
-            'cf_role': 'mesh_topology',
-            'topology_dimension': 2,
-            'long_name': u'Topology data of 2D unstructured mesh'})
+    # TODO: Could be lots of tests here.
+    assert nc_has_variable(ds, 'mesh')
+    assert nc_var_has_attr_vals(ds, 'mesh', {
+        'cf_role': 'mesh_topology',
+        'topology_dimension': 2,
+        'long_name': u'Topology data of 2D unstructured mesh'})
+    ds.close()
 
 
 def test_set_mesh_name():
-    fname = 'temp.nc'
     grid = two_triangles()
-    grid.mesh_name = "mesh_2"
-
-    grid.save_as_netcdf(fname)
-
-    with netCDF4.Dataset(fname) as ds:
-        assert nc_has_variable(ds, 'mesh_2')
-        assert nc_var_has_attr_vals(ds, 'mesh_2', {
-            'cf_role': 'mesh_topology',
-            'topology_dimension': 2,
-            'long_name': u'Topology data of 2D unstructured mesh'})
-        assert nc_var_has_attr_vals(ds, 'mesh_2', {
-            'cf_role': 'mesh_topology',
-            'topology_dimension': 2,
-            'long_name': u'Topology data of 2D unstructured mesh',
-            'node_coordinates': 'mesh_2_node_lon mesh_2_node_lat'})
-
-        assert nc_has_variable(ds, 'mesh_2_node_lon')
-        assert nc_has_variable(ds, 'mesh_2_node_lat')
-        assert nc_has_variable(ds, 'mesh_2_face_nodes')
-        assert nc_has_variable(ds, 'mesh_2_edge_nodes')
+    grid.mesh_name = 'mesh_2'
 
-        assert nc_has_dimension(ds, "mesh_2_num_node")
-        assert nc_has_dimension(ds, "mesh_2_num_edge")
-        assert nc_has_dimension(ds, "mesh_2_num_face")
-        assert nc_has_dimension(ds, "mesh_2_num_vertices")
-
-        assert not nc_var_has_attr(ds, 'mesh_2', "face_edge_connectivity")
+    fname = 'temp.nc'
+    with chdir(test_files):
+        grid.save_as_netcdf(fname)
+        ds = netCDF4.Dataset(fname)
+        os.remove(fname)
+
+    assert nc_has_variable(ds, 'mesh_2')
```

Comment:
I'm a bit confused as to how this works -- you open a DataSet against the file you just created, then delete that file. Is the dataset still good? I think it can be on *nix, but I'd expect Windows to get really confused, what with its aggressive file locking and all. but I guess if it does indeed work on Windows then we're OK.